### ISSUE TITLE
Crash if trying to save without touching the canvas?

### ISSUE THREAD

**Issue by rborn** [jgrancher/react-native-sketch#37]
So If I call save without drawing anything I get a crash

~~~
default	15:33:58.172747 +0100	paack	Exception '*** -[_NSPlaceholderData initWithBase64EncodedString:options:]: nil string argument' was thrown while invoking saveDrawing on target RNSketchManager with params (
    "<null>",
    png,
    2618,
    2619
)
callstack: (
	0   CoreFoundation                      0x00000001058421ab __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x0000000103d3cf41 objc_exception_throw + 48
	2   CoreFoundation                      0x00000001058b6cb5 +[NSException raise:format:] + 197
	3   Foundation                          0x00000001037612bb -[NSData(NSData) initWithBase64EncodedString:options:] + 84
	4   paack                               0x000000010323651a -[RNSketchManager saveDrawing:ofType:resolve:reject:] + 538
	5   CoreFoundation                      0x00000001057c649c __invoking___ + 140
	6   CoreFoundation                      0x00000001057c6370 -[NSInvocation invoke] + 320
	7   CoreFoundation                      0x00000001057ded56 -[NSInvocation invokeWithTarget:] + 54
	8   pa<…>

~~~

Thank you 😊

**Comment by rborn** [comment]
A workaround on this is to intercept the onChange event and enable the save button but it's not ok I think 😄

**Comment by jgrancher** [comment]
@rborn Good point! What do you think should be the best solution?
Was thinking of returning a `Promise.reject('No image provided')` or something like that from `index.ios.js` after a `sketch.save()` call when the `imageData` is null.
That way, it doesn't crash but just send a console warning. Thoughts?

**Comment by rborn** [comment]
@jgrancher yes, I think it's ok. Any "emergency" plan actually it's good, as long as the app doesn't crash :)
Thank you! 🤗

**Comment by rborn** [comment]
Oh, one more thing. I'd go with some error codes for the error, so if there is other thing causing the crash to be able to catch it in the app - like no space on device or some other error the system returns (not sure whet else it could be though)

**Comment by rborn** [comment]
any news o this? 🙌

**Comment by jgrancher** [comment]
@rborn Would you be ok if I merge [this PR](https://github.com/jgrancher/react-native-sketch/pull/38) now? Then we can create another PR later on for error codes. Thoughts?

**Comment by rborn** [comment]
Yes sir, thank you 😊

**Comment by jgrancher** [comment]
Fixed with `1.0.3` just published. Let me know if anything, I'll close this in the meantime!

**Comment by rborn** [comment]
@jgrancher works great, thank you 😸
